stellargraph/layer/appnp.py

```diff
@@ -349,7 +333,26 @@ def __call__(self, x):
 
         return h_layer
 
-    def _tensors(self, multiplicity, extra_layers):
+    def __call__(self, x):
```

Comment:
It looks like `__call__` now just wraps the `_run` function. Could you say why you went for this approach instead of something like adding a `feature_layers` kwarg in `__call__? 
 e.g.

```
def __call__(self, x, feature_layers=None):
    ...
    if feature_layers is None:
        feature_layers = self.feature_layers
    ...
```

Reply:
I didn't think it makes sense for a user to see or use the `feature_layers` argument, so it's better to not provide it at all, rather than having it hanging around and potentially confusing things (and also, if it exists, we have to support it forever, or risk breaking users if we change things). What do you think?

Reply:
Cool, I think that's a good point. The the user shouldnt even need to use the `feature_layers` arg